packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -441,6 +489,32 @@ export class FirestoreRepository implements Repository {
     return customElementsResults.docs.map((d) => d.data());
   }
 
+  async queryElements({
+    query,
+    limit,
+  }: {
+    query?: string;
+    limit?: number;
+  }): Promise<Array<CustomElement>> {
+    let dbQuery: Query<CustomElement> | CollectionGroup<CustomElement> = db
+      .collectionGroup('customElements')
+      .withConverter(customElementConverter)
+      .where('isLatest', '==', true)
+      .limit(limit ?? 25);
+
+    if (query !== undefined) {
+      // Split query
+      const queryTerms = natural.PorterStemmer.tokenizeAndStem(query);
+      if (queryTerms.length > 10) {
+        queryTerms.length = 10;
```

Comment:
Why limit at 10?

Reply:
That's the Firestore limit: https://cloud.google.com/firestore/docs/query-data/queries#array-contains-any

I should make a TODO/issue - If we keep this approach, we want to sort the query by inverse frequency so we keep the most important terms. Then 10 seems like enough. For longer queries I _think_ you get better results with n-gram searches, but that's stretching my knowledge and probably another reason to use a service :)